themes/bootstrap5/templates/myresearch/bulk-action-buttons.phtml

```diff
@@ -9,92 +10,109 @@
             ]
         )
       ?>
-    <?php endif; ?>
-    <?php if ((null !== $this->list && $this->userlist()->userCanEditList($user, $list)) || null === $this->list && $user): ?>
-      <?=
-        $this->bulkAction()->button(
-            'delete',
-            'ui-delete',
-            'Delete',
-            [
-              'id' => $this->idPrefix . 'delete_list_items_' . (null !== $this->list ? $this->escapeHtmlAttr($this->list->getId()) : ''),
-              'title' => $this->translate('bulk_delete_myresearch'),
-            ]
-        )
-      ?>
-    <?php endif; ?>
-    <?php $exportOptions = $this->export()->getActiveFormats('bulk'); ?>
-    <?php if (count($exportOptions) > 0): ?>
-      <?=
-        $this->bulkAction()->button(
-            'export',
-            'export',
-            'Export',
-            [
-              'title' => $this->translate('bulk_export_myresearch'),
-            ]
-        )
-      ?>
-    <?php endif; ?>
-    <?=
-      $this->bulkAction()->button(
-          'print',
-          'print',
-          'Print',
-          [
-            'title' => $this->translate('bulk_print_myresearch'),
-            'data-lightbox-ignore' => '',
-          ]
-      )
-    ?>
-    <?php if ($this->cart()->isActive()): ?>
-      <?=
-        $this->bulkAction()->button(
-            'add',
-            'cart-add',
-            'Add to Book Bag',
-            [
-              'id' => $this->idPrefix . 'updateCart',
-            ]
-        )
-      ?>
-    <?php endif; ?>
-  </ul>
-  <ul class="selection-controls-bar">
-    <?php if(in_array($checkboxSelectAllType, ['on_page', 'both'])): ?>
-      <li class="bulk-checkbox">
-        <label for="myresearchCheckAll">
-          <input type="checkbox" name="selectAll" class="checkbox-select-all" id="myresearchCheckAll">
-          <?=$this->transEsc('select_all_on_page')?>
-        </label>
-      </li>
-    <?php endif; ?>
-    <?php if($multiPageSelection && in_array($checkboxSelectAllType, ['global', 'both'])): ?>
-      <li class="bulk-checkbox">
-        <div class="select-all-global hidden">
-          <label for="myresearchCheckAllGlobal">
-            <input type="checkbox" name="selectAllGlobal" class="checkbox-select-all-global" id="myresearchCheckAllGlobal">
-            <?=$this->transEsc('select_all')?>
+      <?php if ($this->cart()->isActive()): ?>
+        <?=
+          $this->bulkAction()->button(
+              'add',
+              'cart-add',
+              'Add to Book Bag',
+              [
+                'id' => $this->idPrefix . 'updateCart',
+              ]
+          )
+        ?>
+      <?php endif; ?>
+    </ul>
+  <?php
+    $bulkActionButtons = ob_get_contents();
+    ob_end_clean();
+    ob_start();
+  ?>
+    <ul class="selection-controls-bar">
+      <?php if(in_array($checkboxSelectAllType, ['on_page', 'both'])): ?>
+        <li class="bulk-checkbox">
+          <label for="<?=$this->idPrefix?>myresearchCheckAll">
```

Comment:
While I realize the likelihood of an unusual value being passed in the idPrefix property is very low, I wonder if it would be better to more explicitly escape these values. Maybe it's worth considering using the htmlAttributes helper to improve readability... e.g.:

```
<?php $allCheckboxId = $this->idPrefix . 'myresearchCheckAll'; ?>
<label<?=$this->htmlAttributes(['id' => $allCheckboxId])?>>
  <input<?=$this->htmlAttributes(['type' => 'checkbox', 'name' => 'selectAll', 'class' => 'checkbox-select-all', 'id' => $allCheckboxId])?>>
  <?=$this->transEsc('select_all_on_page')?>
</label>
```

I realize that this approach isn't perfect either, and I don't insist on a change... just brainstorming!